src/app.py

```diff
@@ -63,7 +65,30 @@ def change_username():
         )
         os.remove(new_csv_file_path)
         username = new_username
-    return render_template("index.html", username=username)
+    return render_template("index.html", emails=emails, username=username)
+
+
+@app.route("/get_emails", methods=["GET"])
+def extract_emails_from_pdf():
+    global username
```

Comment:
I don't think you need to define `global username` here as you didn't change the value of this global variable inside your function. There is no need to define it if your function didn't modify the value of it